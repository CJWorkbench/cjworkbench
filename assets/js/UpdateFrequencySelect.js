import React from 'react'
import PropTypes from 'prop-types'
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap'
import { Form, FormGroup, Label, Input } from 'reactstrap'
import workbenchapi from './WorkbenchAPI';
import dateFormat from 'dateformat'


export default class UpdateFrequencySelect extends React.Component {
  constructor(props) {
    super(props);
    this.state = { 
      modalOpen: false,
      liveSettings: {
        manual: !this.props.updateSettings.autoUpdateData,
        period: this.props.updateSettings.updateInterval,
        unit: this.props.updateSettings.updateUnits
      },
      dialogSettings: {
        manual: !this.props.updateSettings.autoUpdateData,
        period: this.props.updateSettings.updateInterval,
        unit: this.props.updateSettings.updateUnits
      }
    };

    // Allow props to specify a conversion from browser time to displayed time, so tests can run in UTC (not test machine tz) 
    if (props.timezoneOffset != undefined) {
      this.state.timezoneOffset = props.timezoneOffset;
    } else {
      this.state.timezoneOffset = 0; // display in browser local time 
    }

    this.api = workbenchapi();
    this.toggleModal = this.toggleModal.bind(this);
    this.toggleManual = this.toggleManual.bind(this);  
    this.updatePeriod = this.updatePeriod.bind(this);    
    this.updateUnit = this.updateUnit.bind(this);   
    this.saveSettings = this.saveSettings.bind(this);    
  }

  toggleModal() {
    this.setState({modalOpen: !this.state.modalOpen })
  }

  toggleManual() {
    this.setState(
      {dialogSettings: { 
        manual: !this.state.dialogSettings.manual,
        period: this.state.dialogSettings.period,
        unit: this.state.dialogSettings.unit
    }});
  }

  updatePeriod(event) {
    this.setState(
        {dialogSettings: { 
          manual: this.state.dialogSettings.manual,
          period: event.target.value,
          unit: this.state.dialogSettings.unit
    }});
  }

  updateUnit(event) {
    this.setState(
      {dialogSettings: { 
        manual: this.state.dialogSettings.manual,
        period: this.state.dialogSettings.period,
        unit: event.target.value
    }});
  }

  saveSettings() {
    var params = {
      auto_update_data: !this.state.dialogSettings.manual,
      update_interval: this.state.dialogSettings.period,
      update_units: this.state.dialogSettings.unit
    };
    this.api.setWfModuleUpdateSettings(this.props.wfModuleId, params); 
    this.setState({liveSettings: this.state.dialogSettings});
    this.toggleModal();
  }

  render() {

    //button highlights
    var highlightManual = this.state.dialogSettings.manual ? 'action-button button-full-blue' : 'action-button button-gray';
    var highlightAuto = !this.state.dialogSettings.manual ? 'action-button button-full-blue' : 'action-button button-gray';

    var manualText = this.state.dialogSettings.manual ? 'action-button button-full-blue' : 'action-button button-gray';
    var autoText = !this.state.dialogSettings.manual ? 'action-button button-full-blue' : 'action-button button-gray';
    

    // info shown on Wf Module card
    var manual = this.state.liveSettings.manual;
    var period = this.state.liveSettings.period;
    var unit = this.state.liveSettings.unit;
    var settingsInfo = manual 
      ? 'Manual'
      :'Auto, every ' + period + ' ' + unit;

    var lastChecked = null;
    if (this.props.updateSettings.lastUpdateCheck) 
      lastChecked = <div className='content-3 t-m-gray'>Last checked: {dateFormat(this.props.updateSettings.lastUpdateCheck, "mmmm d yyyy - hh:MM TT")}</div>
      
    return (
      <div className='version-item'>
        <div className='' >
          <span className='content-3 t-d-gray'>Update: </span>
          <span className='content-3 t-f-blue' onClick={this.toggleModal}>{settingsInfo}</span>  
        </div>
        {lastChecked}
        <Modal isOpen={this.state.modalOpen} toggle={this.toggleModal} className='dialog-window'>
          <ModalHeader toggle={this.toggleModal} className='dialog-header'>
            <span className='title-4 t-d-gray'>Sync Settings</span>
          </ModalHeader>
          <ModalBody className='dialog-body'>
            <FormGroup>
              <Label for="updateFreq" className='content-3 t-d-gray mb-2'>Check for update every</Label>
              <div className='update-freq-settings update-freq-test-class mb-2'>
                <Input 
                  type="number" 
                  onChange={this.updatePeriod}
                  value={this.state.dialogSettings.period}
                  min='1' 
                  max='500' 
                  name="updateFreq" 
                  id="updateFreqNum">
                </Input>
                <Input 
                  type="select" 
                  value={this.state.dialogSettings.unit}
                  onChange={this.updateUnit}
                  name="updateFreq" 
                  id="updateFreqPeriod" 
                >
                  <option>seconds</option>
                  <option>minutes</option>
                  <option>hours</option>                
                  <option>days</option>
                  <option>weeks</option>
                </Input>
              </div>
              <div className='content-3 t-d-gray mb-2'>When an update is found:</div>
              <div className="d-flex flex-row mb-2">
                <div onClick={this.toggleManual} className={highlightManual} >Manual</div>
                <div className='t-d-gray info-1'>Notify me, but do not update the workflow. (recommended)</div>   
              </div>    
              <div className="d-flex flex-row mb-2">                     
                <div onClick={this.toggleManual} className={highlightAuto} >Auto</div>
                <div className='t-d-gray info-1'>Automatically save the current version of the workflow and update it with the newest data.</div>  
              </div>            
            </FormGroup>
          </ModalBody>
          <ModalFooter className='dialog-footer'>
            <div className='action-button button-blue' onClick={this.saveSettings}>OK</div>
            <div className='action-button button-blue' onClick={this.toggleModal}>Cancel</div>            
          </ModalFooter>
        </Modal>
      </div>
    );
  }

}

UpdateFrequencySelect.propTypes = {
  updateSettings:   PropTypes.object.isRequired,
  wfModuleId:       PropTypes.number.isRequired
};
