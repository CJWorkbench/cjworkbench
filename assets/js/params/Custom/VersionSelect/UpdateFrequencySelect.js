import React from 'react'
import PropTypes from 'prop-types'
import UpdateFrequencySelectModal from './UpdateFrequencySelectModal'
import { timeDifference } from '../../../utils'
import { trySetWfModuleAutofetchAction, setWfModuleNotificationsAction } from '../../../workflow-reducer'
import { connect } from 'react-redux'
import { Trans, t } from '@lingui/macro'
import { withI18n } from '@lingui/react'

export const UpdateFrequencySelect = withI18n()(class UpdateFrequencySelect extends React.PureComponent {
  static propTypes = {
    i18n: PropTypes.shape({
      // i18n object injected by LinguiJS withI18n()
      _: PropTypes.func.isRequired
    }),
    workflowId: PropTypes.number.isRequired,
    wfModuleId: PropTypes.number.isRequired,
    isAnonymous: PropTypes.bool.isRequired,
    isReadOnly: PropTypes.bool.isRequired,
    lastCheckDate: PropTypes.instanceOf(Date), // null if never updated
    isAutofetch: PropTypes.bool.isRequired,
    fetchInterval: PropTypes.number.isRequired,
    isEmailUpdates: PropTypes.bool.isRequired,
    setEmailUpdates: PropTypes.func.isRequired, // func(wfModuleId, isEmailUpdates) => undefined
    trySetAutofetch: PropTypes.func.isRequired // func(wfModuleId, isAutofetch, fetchInterval) => Promise[response]
  }

  state = {
    isModalOpen: false,
    quotaExceeded: null // JSON response -- contains autofetch info iff we exceeded quota
  }

  handleClickOpenModal = (ev) => {
    if (ev && ev.preventDefault) ev.preventDefault() // <a> => do not change URL
    if (this.props.isReadOnly) return
    if (this.props.isAnonymous) return

    this.setState({
      isModalOpen: true
    })
  }

  handleCloseModal = () => {
    this.setState({
      isModalOpen: false
    })
  }

  setEmailUpdates = (isEmailUpdates) => {
    const { setEmailUpdates, wfModuleId } = this.props
    setEmailUpdates(wfModuleId, isEmailUpdates)
  }

  trySetAutofetch = (isAutofetch, fetchInterval) => {
    const { trySetAutofetch, wfModuleId } = this.props
    return trySetAutofetch(wfModuleId, isAutofetch, fetchInterval)
  }

  render () {
    const { i18n, lastCheckDate, isAutofetch, fetchInterval, isEmailUpdates, workflowId, wfModuleId } = this.props
    const { isModalOpen } = this.state

    return (
      <div className='update-frequency-select'>
        <div className='update-option'>
          <span className='version-box-option'><Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.update'>Update</Trans> </span>
          <a
            href='#'
            title={i18n._(t('js.params.Custom.VersionSelect.UpdateFrequencySelect.changeUpdateSettings.hoverText')`change auto-update settings`)}
            className='content-1 ml-1 action-link'
            onClick={this.handleClickOpenModal}
          >
            {isAutofetch ? <Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.auto'>Auto</Trans> : <Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.manual'>Manual</Trans>}
          </a>
        </div>
        {lastCheckDate ? (
          <div className='last-checked'>
            <Trans id='js.params.Custom.VersionSelect.UpdateFrequencySelect.lastChecked' description="The parameter is a time difference (i.e. something like '4h ago'. The tag is a <time> tag.">
                Checked <time dateTime={this.props.lastCheckDate.toISOString()}>{timeDifference(lastCheckDate, Date.now(), i18n)}</time>
            </Trans>
          </div>
        ) : null}
        {isModalOpen ? (
          <UpdateFrequencySelectModal
            workflowId={workflowId}
            wfModuleId={wfModuleId}
            isEmailUpdates={isEmailUpdates}
            isAutofetch={isAutofetch}
            fetchInterval={fetchInterval}
            setEmailUpdates={this.setEmailUpdates}
            trySetAutofetch={this.trySetAutofetch}
            onClose={this.handleCloseModal}
          />
        ) : null}
      </div>
    )
  }
})

const mapStateToProps = (state, ownProps) => {
  const workflow = state.workflow || {}
  const wfModule = state.wfModules[String(ownProps.wfModuleId)] || {}
  // We need a "default" value for everything: wfModule might be a placeholder

  const lastCheckString = wfModule.last_update_check // JSON has no date -- that's a STring
  const lastCheckDate = lastCheckString ? new Date(Date.parse(lastCheckString)) : null

  return {
    lastCheckDate,
    workflowId: workflow.id,
    isReadOnly: workflow.read_only,
    isAnonymous: workflow.is_anonymous,
    isEmailUpdates: wfModule.notifications || false,
    isAutofetch: wfModule.auto_update_data || false,
    fetchInterval: wfModule.update_interval || 86400
  }
}

const mapDispatchToProps = {
  trySetAutofetch: trySetWfModuleAutofetchAction,
  setEmailUpdates: setWfModuleNotificationsAction
}

export default connect(mapStateToProps, mapDispatchToProps)(UpdateFrequencySelect)
