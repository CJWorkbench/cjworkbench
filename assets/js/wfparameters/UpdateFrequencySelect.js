import React from 'react'
import PropTypes from 'prop-types'
import UpdateFrequencySelectModal from './UpdateFrequencySelectModal'
import { timeDifference } from '../utils'
import { updateWfModuleAction } from '../workflow-reducer'
import { connect } from 'react-redux'

export class UpdateFrequencySelect extends React.PureComponent {
  static propTypes = {
    wfModuleId: PropTypes.number.isRequired,
    isAnonymous: PropTypes.bool.isRequired,
    isReadOnly: PropTypes.bool.isRequired,
    lastCheckDate: PropTypes.instanceOf(Date), // null if never updated
    settings: PropTypes.shape({
      isAutoUpdate: PropTypes.bool.isRequired,
      isEmailUpdates: PropTypes.bool.isRequired,
      timeNumber: PropTypes.number.isRequired,
      timeUnit: PropTypes.oneOf([ 'minutes', 'hours', 'days', 'weeks' ]).isRequired,
    }).isRequired,
    updateSettings: PropTypes.func.isRequired, // func({ isAutoUpdate, isEmailUpdates, timeNumber, timeUnit }) -> undefined
  }

  constructor(props) {
    super(props);
    this.state = {
      isModalOpen: false,
    }
  }

  onOpenModal = (ev) => {
    if (ev && ev.preventDefault) ev.preventDefault() // <a> => do not change URL
    if (this.props.isReadOnly) return
    if (this.props.isAnonymous) return

    this.setState({
      isModalOpen: true,
    })
  }

  onSubmit = (settings) => {
    this.props.updateSettings(settings)
    // TODO keep modal open until server responds with OK?
    this.setState({
      isModalOpen: false,
    })
  }

  onCancel = () => {
    this.setState({
      isModalOpen: false,
    })
  }

  render() {
    const lastChecked = this.props.lastCheckDate ? (
      <div className="content-4 t-m-gray">
        Checked <time time={this.props.lastCheckDate.toISOString()}>{timeDifference(this.props.lastCheckDate, Date.now())}</time>
      </div>
    ) : null

    const autoOrManual = this.props.settings.isAutoUpdate ? 'Auto' : 'Manual'

    const maybeModal = this.state.isModalOpen ? (
        <UpdateFrequencySelectModal
          {...this.props.settings}
          onCancel={this.onCancel}
          onSubmit={this.onSubmit}
          />
    ) : null

    return (
      <div className='frequency-item'>
        <div>
          <span className='content-3 t-d-gray'>Update </span>
          <a href="#" title="change auto-update settings" className='content-3 ml-1 action-link' onClick={this.onOpenModal}>{autoOrManual}</a>
        </div>
        {lastChecked}
        {maybeModal}
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  const workflow = state.workflow || {}
  const wfModules = workflow.wf_modules || {}
  const wfModule = wfModules.find(wfm => wfm.id === ownProps.wfModuleId) || {}
  // We need a "default" value for everything: wfModule might be a placeholder

  const lastCheckString = wfModule.last_update_check // JSON has no date -- that's a STring
  const lastCheckDate = lastCheckString ? new Date(Date.parse(lastCheckString)) : null

  return {
    lastCheckDate,
    isReadOnly: workflow.read_only,
    isAnonymous: workflow.is_anonymous,
    settings: {
      isAutoUpdate: wfModule.auto_update_data || false,
      isEmailUpdates: wfModule.notifications || false,
      timeNumber: wfModule.update_interval || 1,
      timeUnit: wfModule.update_units || 'days',
    }
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    updateSettings: (settings) => {
      const action = updateWfModuleAction(ownProps.wfModuleId, {
        auto_update_data: settings.isAutoUpdate,
        update_interval: settings.timeNumber,
        update_units: settings.timeUnit,
        notifications: settings.isEmailUpdates,
      })
      dispatch(action)
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(UpdateFrequencySelect)
