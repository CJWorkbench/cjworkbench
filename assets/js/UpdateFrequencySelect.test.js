import React from 'react'
import UpdateFrequencySelect  from './UpdateFrequencySelect'
import { mount } from 'enzyme'

it('UpdateFrequencySelect renders correctly', () => {

  const wrapper = mount(<UpdateFrequencySelect
    updateSettings= {{
      lastUpdateCheck: 2,
      autoUpdateData: true,
      updateInterval: 5,
      updateUnits: 'minutes'
    }}
    wfModuleId={1}
   />);
  expect(wrapper).toMatchSnapshot();

  wrapper.setState( { modalOpen: true } )
  expect(wrapper).toMatchSnapshot();

});




