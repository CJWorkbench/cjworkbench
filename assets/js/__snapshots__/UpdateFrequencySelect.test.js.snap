// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateFrequencySelect Does not save changed settings when user hits Cancel 1`] = `
<div
  className="version-item"
>
  <div
    className="mb-4"
  >
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      manual
    </span>
  </div>
  <div
    className="content-3 t-m-gray"
  >
    Checked 
    2 days ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="dialog-window"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        Sync Settings
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <Label
          className="content-3 t-d-gray mb-3"
          for="updateFreq"
          tag="label"
        >
          Check for update every
        </Label>
        <div
          className="update-freq-settings update-freq-test-class mb-5"
        >
          <Input
            className="input-number mr-4"
            id="updateFreqNum"
            max="500"
            min="1"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="number"
            value={888}
          />
          <Input
            className="input-dropdown"
            id="updateFreqUnit"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="select"
            value="days"
          >
            <option>
              seconds
            </option>
            <option>
              minutes
            </option>
            <option>
              hours
            </option>
            <option
              id="days-option"
            >
              days
            </option>
            <option>
              weeks
            </option>
          </Input>
        </div>
        <div
          className="content-3 t-d-gray mb-3"
        >
          When an update is found:
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button manual-button button-gray "
            onClick={[Function]}
          >
            Manual
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Notify me, but do not update the workflow. (recommended)
          </div>
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button auto-button button-full-blue"
            onClick={[Function]}
          >
            Auto
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Automatically save the current version of the workflow and update it with the newest data.
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        OK
      </div>
      <div
        className="action-button button-blue test-cancel-button"
        onClick={[Function]}
      >
        Cancel
      </div>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`UpdateFrequencySelect Renders correctly when in Private mode and changes are confirmed with OK 1`] = `
<div
  className="version-item"
>
  <div
    className="mb-4"
  >
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      manual
    </span>
  </div>
  <div
    className="content-3 t-m-gray"
  >
    Checked 
    2 days ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="dialog-window"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        Sync Settings
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <Label
          className="content-3 t-d-gray mb-3"
          for="updateFreq"
          tag="label"
        >
          Check for update every
        </Label>
        <div
          className="update-freq-settings update-freq-test-class mb-5"
        >
          <Input
            className="input-number mr-4"
            id="updateFreqNum"
            max="500"
            min="1"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="number"
            value={5}
          />
          <Input
            className="input-dropdown"
            id="updateFreqUnit"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="select"
            value="minutes"
          >
            <option>
              seconds
            </option>
            <option>
              minutes
            </option>
            <option>
              hours
            </option>
            <option
              id="days-option"
            >
              days
            </option>
            <option>
              weeks
            </option>
          </Input>
        </div>
        <div
          className="content-3 t-d-gray mb-3"
        >
          When an update is found:
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button manual-button button-full-blue"
            onClick={[Function]}
          >
            Manual
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Notify me, but do not update the workflow. (recommended)
          </div>
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button auto-button button-gray"
            onClick={[Function]}
          >
            Auto
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Automatically save the current version of the workflow and update it with the newest data.
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        OK
      </div>
      <div
        className="action-button button-blue test-cancel-button"
        onClick={[Function]}
      >
        Cancel
      </div>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`UpdateFrequencySelect Renders correctly when in Private mode and changes are confirmed with OK 2`] = `
<div
  className="version-item"
>
  <div
    className="mb-4"
  >
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      manual
    </span>
  </div>
  <div
    className="content-3 t-m-gray"
  >
    Checked 
    2 days ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="dialog-window"
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        Sync Settings
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <Label
          className="content-3 t-d-gray mb-3"
          for="updateFreq"
          tag="label"
        >
          Check for update every
        </Label>
        <div
          className="update-freq-settings update-freq-test-class mb-5"
        >
          <Input
            className="input-number mr-4"
            id="updateFreqNum"
            max="500"
            min="1"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="number"
            value={5}
          />
          <Input
            className="input-dropdown"
            id="updateFreqUnit"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="select"
            value="minutes"
          >
            <option>
              seconds
            </option>
            <option>
              minutes
            </option>
            <option>
              hours
            </option>
            <option
              id="days-option"
            >
              days
            </option>
            <option>
              weeks
            </option>
          </Input>
        </div>
        <div
          className="content-3 t-d-gray mb-3"
        >
          When an update is found:
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button manual-button button-full-blue"
            onClick={[Function]}
          >
            Manual
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Notify me, but do not update the workflow. (recommended)
          </div>
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button auto-button button-gray"
            onClick={[Function]}
          >
            Auto
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Automatically save the current version of the workflow and update it with the newest data.
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        OK
      </div>
      <div
        className="action-button button-blue test-cancel-button"
        onClick={[Function]}
      >
        Cancel
      </div>
    </ModalFooter>
  </Modal>
</div>
`;

exports[`UpdateFrequencySelect Renders correctly when in Private mode and changes are confirmed with OK 3`] = `
<div
  className="version-item"
>
  <div
    className="mb-4"
  >
    <span
      className="content-3 t-d-gray"
    >
      Update 
    </span>
    <span
      className="content-3 t-f-blue test-modal-button"
      onClick={[Function]}
    >
      auto
    </span>
  </div>
  <div
    className="content-3 t-m-gray"
  >
    Checked 
    2 days ago
  </div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    className="dialog-window"
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransitionTimeout={300}
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      className="dialog-header"
      tag="h4"
      toggle={[Function]}
      wrapTag="div"
    >
      <span
        className="title-4 t-d-gray"
      >
        Sync Settings
      </span>
    </ModalHeader>
    <ModalBody
      className="dialog-body"
      tag="div"
    >
      <FormGroup
        tag="div"
      >
        <Label
          className="content-3 t-d-gray mb-3"
          for="updateFreq"
          tag="label"
        >
          Check for update every
        </Label>
        <div
          className="update-freq-settings update-freq-test-class mb-5"
        >
          <Input
            className="input-number mr-4"
            id="updateFreqNum"
            max="500"
            min="1"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="number"
            value={888}
          />
          <Input
            className="input-dropdown"
            id="updateFreqUnit"
            name="updateFreq"
            onChange={[Function]}
            tag="p"
            type="select"
            value="days"
          >
            <option>
              seconds
            </option>
            <option>
              minutes
            </option>
            <option>
              hours
            </option>
            <option
              id="days-option"
            >
              days
            </option>
            <option>
              weeks
            </option>
          </Input>
        </div>
        <div
          className="content-3 t-d-gray mb-3"
        >
          When an update is found:
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button manual-button button-gray "
            onClick={[Function]}
          >
            Manual
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Notify me, but do not update the workflow. (recommended)
          </div>
        </div>
        <div
          className="d-flex flex-row mb-4"
        >
          <div
            className="action-button auto-button button-full-blue"
            onClick={[Function]}
          >
            Auto
          </div>
          <div
            className="t-d-gray info-1 ml-4"
          >
            Automatically save the current version of the workflow and update it with the newest data.
          </div>
        </div>
      </FormGroup>
    </ModalBody>
    <ModalFooter
      className="dialog-footer"
      tag="div"
    >
      <div
        className="action-button button-blue test-ok-button"
        onClick={[Function]}
      >
        OK
      </div>
      <div
        className="action-button button-blue test-cancel-button"
        onClick={[Function]}
      >
        Cancel
      </div>
    </ModalFooter>
  </Modal>
</div>
`;
